packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -366,6 +394,26 @@ class AndroidCameraCameraX extends CameraPlatform {
     previewInitiallyBound = true;
     _previewIsPaused = false;
 
+    // Retrieve info required for correcting the rotation of the camera preview
+    // if necessary.
+
+    final Camera2CameraInfo camera2CameraInfo =
+        await proxy.getCamera2CameraInfo(cameraInfo!);
+    await Future.wait(<Future<Object>>[
+      SystemServices.isPreviewPreTransformed()
+          .then((bool value) => isPreviewPreTransformed = value),
+      proxy
+          .getSensorOrientation(camera2CameraInfo)
+          .then((int value) => sensorOrientation = value),
+      proxy
+          .getUiOrientation()
+          .then((DeviceOrientation value) => naturalOrientation ??= value),
+    ]);
+    _subscriptionForDeviceOrientationChanges = onDeviceOrientationChanged()
+        .listen((DeviceOrientationChangedEvent event) {
+      currentDeviceOrientation = event.orientation;
+    });
```

Comment:
@matanlurey if you have a suggestion as to how to make these run in parallel, do let me know!

Reply:
Ah I didn't notice you already had Future.wait, honestly this is fine. LGTM.